### ISSUE TITLE
Adding new `stripped` flag for %in in WebminCore

### ISSUE THREAD

**Issue by qooob** [webmin/webmin#458]
Hi, Jamie.

As I'm working more and more on the delivering data using ajax calls (like recent [Command Line](https://github.com/qooob/authentic-theme/issues/566) enhancement), could you please add to the global  usage the ability to return stripped HTML disregard of `theme.pl` file (not processing it at all).

For example, if queried `/shell/index.cgi?stripped=1`, then WebminCore, should avoid any theme processing, embedding files (header and footer) and other _slow stuff_ and just return raw data?

It would make UI greater.

**Comment by jcameron** [comment]
Perhaps a better solution would be for modules to offer an API mode that returns just XML or something? Even with the theme stuff removed, you would still face the problem of parsing HTML that might change.

**Comment by qooob** [comment]
Well, yes, changing HTML would be harder. For example, shell module always return only one `pre` tag and the same HTML. XML would require extra work in some cases. 

It would be great to have switches, like 1,2,3 - HTML, XML, JSON. However, the easiest would be only HTML for now.

**Comment by jcameron** [comment]
I'd actually prefer to go with XML or JSON output (if that works for you), rather than HTML.

**Comment by qooob** [comment]
Sure, JSON would be great.

**Comment by jcameron** [comment]
What specific operations do you need to be able to perform? Just executing a shell command and getting back the output? Can you handle the history in the JS?

**Comment by qooob** [comment]
I could do history only in JS but I just managed to make theme use server side history from shell module. I know what you mean.

I managed to extract HTML from modules the same fast as in gray theme by bypassing most of the heavy processes on the theme side. For example it takes now 200ms for /etc folder and before it was 350ms.

The best thing would be to provide JSON support for each module (globally in WebminCore), that would let extract nessesary data. In example of Shell module, we would need to extract: pre tag, pwd, history. 

For mail module (next in a line and very soon): set of control buttons, selects, list of messages (with our without preview based on settings) and etc.

Can this be done for 1.830?
